protocol-definitions/Experimental.yaml

```diff
@@ -1,3 +1,31 @@
 id: 253
 name: Experimental
-methods: []
\ No newline at end of file
+methods:
+  - id: 1
+    name: pipelineSubmit
+    since: 2.7
+    doc: 'The message is used to transfer the declarative pipeline definition and the related resource files 
+    from client to the server.'
+    request:
+      retryable: true
+      partitionIdentifier: -1
+      params:
+        - name: pipelineDefinition
+          type: String
+          nullable: false
+          since: 2.7
+          doc: 'The definition of the pipeline steps. It currently uses the YAML format.'
+        - name: resourceBundle
```

Comment:
Is this for one resource, or several resources? In some situations we may want to provide the job with *two* resources: the "user code" (ie executables) and the "secrets" (needed to connect back to the cluster) which may come from different locations when submitting. Or maybe CLC would aggregate them in one single payload?

What is the max. supported size of the resourceBundle?

Reply:
Max. supported via codec will be configurable system wide with a default of 10MB (see the TDD). User can have larger project but then they need to just use their own custom image API but no project code upload.

We discussed multiple resources case, and it was said that we if for example you have multiple Python steps, each may have different module name and they can all be represented inside the project folder, may be a different python file or may be a different subfolder. Hence, the idea is have every code you need inside this zip and let the resources differentiate via the path/file/module names.

Let me know if you see any problem with this approach and we can discuss.

Reply:
Problem is I don't understand why we inflict this onto ourselves when the normal Jet submit has already figured it out.

Reply:
We utilize JobConfig.attachFile API. Currently, for Python solution all we need is to attach the user's Python project files, hence, we did not see the need for multiple files. But if there is a need surely, we can consider that. Please provide a valid use case which can not be satisfied with the current approach?